### PR TITLE
Report failing tests without `//~ ERROR` comments

### PR DESCRIPTION
**Pull request by oli-obk**
r? @nikomatsakis

### CODE REVIEW

File: src/tools/compiletest/src/runtest.rs [hunk 1]
```diff
@@ -2781,12 +2781,14 @@ impl<'test> TestCx<'test> {
                explicit, self.config.compare_mode, expected_errors, proc_res.status,
                self.props.error_patterns);
         if !explicit && self.config.compare_mode.is_none() {
-            if !expected_errors.is_empty() && !proc_res.status.success() {
-                // "//~ERROR comments"
-                self.check_expected_errors(expected_errors, &proc_res);
-            } else if !self.props.error_patterns.is_empty() && !proc_res.status.success() {
-                // "// error-pattern" comments
-                self.check_error_patterns(&proc_res.stderr, &proc_res);
+            if !proc_res.status.success() {
```
Comment (nikomatsakis) [inline src/tools/compiletest/src/runtest.rs]: Surely this will require updating various test files?

**Review by nikomatsakis — CHANGES_REQUESTED** [review]
r=me once travis is happy =)

File: src/test/ui/error-codes/E0161.rs [hunk 2]
```diff
@@ -29,7 +29,15 @@
...
-    box *x; //~ ERROR E0161
+    box *x;
+    //[ast]~^ ERROR E0161
+    //[nll]~^^ ERROR E0161
+    //[zflags]~^^^ ERROR E0161
+    //[edition]~^^^^ ERROR E0161
+    //[astul]~^^^^^ ERROR E0161
+    //[nllul]~^^^^^^ ERROR E0161
+    //[zflagsul]~^^^^^^^ ERROR E0161
+    //[editionul]~^^^^^^^^ ERROR E0161
```
Comment (pnkfelix) [inline src/test/ui/error-codes/E0161.rs]: heh. We really need to implement something where a `//~ ERROR` in a revisions test means "this applies to all revisons" ...

Comment (petrochenkov) [reply src/test/ui/error-codes/E0161.rs]: @oli-obk 
Why, `^^^^^^`, `//~| ERROR` doesn't work with versions?

Comment (oli-obk) [reply src/test/ui/error-codes/E0161.rs]: nope

Comment (nikomatsakis) [reply src/test/ui/error-codes/E0161.rs]: yeah I wish it did

Comment (nikomatsakis) [reply src/test/ui/error-codes/E0161.rs]: I was just too lazy to fix it

**Review by nikomatsakis — APPROVED** [review]

**Comment by nikomatsakis** [comment]
Getting close! 🎉

**Comment by kennytm** [comment]
@bors p=4

Raising priority as this may affect / be affected by many other PRs.

**Pull request merged**